packages/mermaid/src/populateDb.ts

```diff
@@ -0,0 +1,15 @@
+import type { Common } from 'mermaid-parser';
+import { isInfo } from 'mermaid-parser';
+import type { DiagramDB } from './diagram-api/types.js';
+import { InfoDB } from './diagrams/info/infoTypes.js';
+
+export function populateDb(ast: Common, db: DiagramDB) {
+  ast.accDescr && db.setAccDescription?.(ast.accDescr);
+  ast.accTitle && db.setAccTitle?.(ast.accTitle);
+  ast.title && db.setDiagramTitle?.(ast.title);
+
+  if (isInfo(ast)) {
+    const infoDb = db as InfoDB;
+    ast.version && infoDb.setVersion(ast.version);
+  }
+}
```

Comment:
This should be inside the individual diagram folders. This will remove the need for `isInfo` function as we already know what type the diagram is (as it will be called after the parse function)
The common functions can be in a common file. 

```
  ast.accDescr && db.setAccDescription?.(ast.accDescr);
  ast.accTitle && db.setAccTitle?.(ast.accTitle);
  ast.title && db.setDiagramTitle?.(ast.title);
```

Reply:
But why `commonDb` is inside `packages/mermaid/src/commonDb.ts`

Reply:
I didn't understand the question 😅

Reply:
And I'm thinking of setting some default values to `undefined`. I think it would make sense (that's what actually happens in langium):


- undefined === not parsed
- empty string === parsed but ignored in setting values
- string === set values

Reply:
> I didn't understand the question 😅

I mean why it's inside `src` instead of `stc/diagrams/common`.

Reply:
> I don't know. Maybe the diagrams/common didn't exist back when it was introduced.

Should we move it there? Or another PR? It might need some refactoring also.

> Btw, by `the individual diagram folders`, I meant `diagrams/info/populateDB.ts`. The populateDB function inside `diagrams/info` will have the logic to populate the infoDB from the info AST. No need to use ` if (isInfo(ast)) {` at all.
> 
> You can call a common `populateCommonDB` with the AST to set the 3 common values to the common db.

But in that case we will have two DB files in the same diagram. And I think we follow naming convention in diagram files, the diagram word as prefix, naming it `populateDB` isn't following that.

Reply:
> Should we move it there? Or another PR? It might need some refactoring also.

Not now. 

> But in that case we will have two DB files in the same diagram. And I think we follow naming convention in diagram files, the diagram word as prefix, naming it populateDB isn't following that.

You can add the populateDB function in the diagramParser.ts file (in mermaid package) then? As the previous parser was doing the parsing and population of DB.

Reply:
Done f5c33c3f1